src/libAtomVM/bif.c

```diff
@@ -154,6 +154,24 @@ term bif_erlang_is_tuple_1(Context *ctx, term arg1)
     return term_is_tuple(arg1) ? TRUE_ATOM : FALSE_ATOM;
 }
 
+term bif_erlang_is_map_1(Context *ctx, term arg1)
+{
+    UNUSED(ctx);
+
+    return term_is_map(arg1) ? TRUE_ATOM : FALSE_ATOM;
+}
+
+term bif_erlang_is_map_key_2(Context *ctx, term arg1, term arg2)
+{
+    UNUSED(ctx);
+
+    if (!term_is_map(arg2)) {
+        RAISE_ERROR(BADMAP_ATOM);
```

Comment:
I think that the right error is `{badmap,{}}`.

Reply:
Nice catch.  I overlooked that.

Unfortunately, the allocation of a tuple may result in a GC, which in turn invalidates the input term arguments.

The only way I see around this (given the current implementation) would be to externalize the input term into a buffer, write the term to the buffer, GC, and then deserialize the buffered term.  That seems like a bit of overkill, so (for now) I am just returning the atom `unsupported` in the second position of the term.

Please advise if you have any thoughts about how best to work around the GC issue.

Reply:
is this a gc bif? anyway I'm ok with your solution, but please, let's comment it with a TODO which explains why and what we should do.